### ISSUE TITLE
Name resolution seems broken in gccgo

### ISSUE THREAD

**Issue by calmh** [TritonDataCenter/pkgsrc-legacy#67]
Attempting to look up a DNS name results in an index out of range error from the standard library.

test.go:

``` go
package main

import (
        "net"
)

func main() {
        println("Testing a TCP connection to an IP")
        test("173.194.113.177:80")
        println("Testing a TCP connection to a DNS name")
        test("www.google.com:80")
        println("Done")
}

func test(addr string) {
        c, err := net.Dial("tcp", addr)
        if err != nil { panic(err) }
        c.Close()
}
```

result:

```
jb@zlogin2:~ $ gccgo -o test test.go 
jb@zlogin2:~ $ ./test
Testing a TCP connection to an IP
Testing a TCP connection to a DNS name
panic: runtime error: index out of range

goroutine 1 [running]:

goroutine 2 [syscall]:
no stack trace available

goroutine 3 [syscall]:
no stack trace available
[2]
jb@zlogin2:~ $ 
```

**Comment by mamash** [comment]
Erroneously closed by Github.

**Comment by mamash** [comment]
Erroneous Github auto-close.

**Comment by jperkin** [comment]
Sorry it's taken so long to get to this ticket.  I tested this in 2016Q1 with both GCC 4.7 and 4.9, and it appears that this issue has been resolved by changes in GCC since 4.7.
- GCC 4.7

```
$ /opt/local/gcc47/bin/gccgo test.go -o test-47
$ ./test-47
Testing a TCP connection to an IP
Testing a TCP connection to a DNS name
panic: runtime error: index out of range

goroutine 1 [running]:

goroutine 2 [syscall]:
no stack trace available

goroutine 3 [syscall]:
no stack trace available
```
- GCC 4.9

```
$ /opt/local/gcc49/bin/gccgo test.go -o test-49
$ ./test-49
Testing a TCP connection to an IP
Testing a TCP connection to a DNS name
Done
```

We also of course now have native go as an option for 64-bit environments.

**Comment by calmh** [comment]
Indeed. Have been using "gc" Go for a good long while now. Thanks anyway. :)
